homeassistant/components/statistics/sensor.py

```diff
@@ -375,25 +379,26 @@ def _add_state_to_queue(self, new_state: State) -> None:
     def _derive_unit_of_measurement(self, new_state: State) -> str | None:
         base_unit: str | None = new_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
         unit: str | None
-        if self.is_binary and self._state_characteristic in STATS_BINARY_PERCENTAGE:
+        state_characteristic = self._state_characteristic
```

Comment:
Why? This just unnecessarily introduces noise. I would understand if you intended to shorten the variable name or for postprocessing. But as it looks right now, this is not an improvement.

Reply:
Next step would be to convert it to a dict lookup to avoid the linear search but was doing it in smaller chunks

Reply:
Doesn't sound like a bad idea :) These various lists developed over time and served me well. If you can think of a more efficient way, looking forward to it!